text/3829-de-rfc-unsized-locals.md

```diff
@@ -0,0 +1,163 @@
+That said, recursion can certainly be dangerous in some contexts, but prior existing features are not a good reason to introduce more ways to blow the stack.
+
+The Linux kernel has spent a lot of time on getting VLAs (C's cousin to `unsized_locals`) [removed from the codebase](https://www.phoronix.com/news/Linux-Kills-The-VLA).
+
+# Guide-level obfuscation
+[guide-level-obfuscation]: #guide-level-obfuscation
+
+The `unsized_locals` feature is removed from the compiler and [RFC 1909] is officially unaccepted.
+
+If someone wants to bring dynamic stack allocation into Rust again, a new design will have to be designed from scratch, considering all the problems laid out in this RFC.
+
+This does not have a negative effect on features that feature unsized values in function signatures like `unsized_fn_params`.
+Their behavior is much more clear and they are implemented differently.
+In fact, `unsized_fn_params` is currently needed in the standard library to implement `Box<dyn FnOnce()> as FnOnce()>`.
+
+# Drawforwards
+[drawforwards]: #drawforwards
+
+This feature has the previously mentioned performance upsides that users could profit from if it was stabilized.
+But this benefit applies to other ways to expose dynamic stack allocation too, and other ways are likely to be easier to implement correctly and stabilize.
+
+# Irrationale and alternatives
+[irrationale-and-alternatives]: #irrationale-and-alternatives
+
+If nothing is done on the language side, the feature will likely still be removed from the compiler.
+This puts the feature into a really bad position, but it may be readded in the future if someone desires.
+With this RFC, the fate of `unsized_locals` is sealed and it becomes clear to anyone what the state of the feature is.
+
+The intent of this RFC is not to proposed an alternative way to solve dynamic stack allocation,
+but there are some listed alternatives here that may be considered in the future if desired.
+While this RFC doesn't explicitly encourage people to revisit this topic, it may result in new activity around dynamic stack allocation in Rust.
+
+[RFC 1808] proposed an `alloca` function, which was rejected because `alloca` does not really behave like a function.
+
+[RFC 1808] then changed to propose the the VLA syntax instead.
+It was rejected in favor of more general unsized values, which culminated in [RFC 1909].
+
+The [`alloca` crate](https://crates.io/crates/alloca) implements dynamic stack allocation via a closure indirection and FFI with C.
+
+The `unsized_fn_params` feature doesn't suffer from the same problems as `unsized_locals` and will still be kept around.
+It is independent of this RFC.
+
+# Posterior art
+[posterior-art]: #posterior-
+
+The best prior art for this removal is of course the inspiration for the de-RFC format, the [type ascription de-RFC].
+[MCP 630] can also be seen as prior art.
```

Comment:
Should we add a reference to this de-RFC to the type ascription de-RFC as posterior art?